libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -29,18 +29,27 @@ fun NavigationRoute.nativeRoute(): RouteInterface = this.nativeRoute
  */
 fun NavigationRoute.refreshRoute(
     initialLegIndex: Int,
+    currentLegGeometryIndex: Int?,
     legAnnotations: List<LegAnnotation?>?,
     incidents: List<List<Incident>?>?,
 ): NavigationRoute {
     val updateLegs = directionsRoute.legs()?.mapIndexed { index, routeLeg ->
         if (index < initialLegIndex) {
             routeLeg
         } else {
-            routeLeg.toBuilder().annotation(
-                legAnnotations?.getOrNull(index)
-            ).incidents(
-                incidents?.getOrNull(index)
-            ).build()
+            val newAnnotation = legAnnotations?.getOrNull(index)
+            val mergedAnnotation =
+                if (index == initialLegIndex && currentLegGeometryIndex != null) {
+                    AnnotationsRefresher.getRefreshedAnnotations(
+                        routeLeg.annotation(),
+                        newAnnotation,
+                        currentLegGeometryIndex
+                    )
+                } else { newAnnotation }
```

Comment:
This replaces the old annotations for upcoming legs with new annotations, however, with the new backed logic couldn't that produce incorrect results?

I'm thinking of two scenarios for routes where backend decided to trim the returned annotations.

First, when the current leg is very long and gets trimmed. Imagine this original annotations (pseudo):
```
route: { first leg: [1, 2, 3], second leg: [4, 5, 6]}
```
We request the refresh starting from the begging (`currentLegGeometryIndex == 0`) and the backend decides to trim the annotations and returns:
```
route: { first leg: [1, 2], second leg: []}
```
In this case we'd correctly refresh the first leg, but just clear the annotations for the second leg. That doesn't seem correct.

Second scenario is even worse. We again request the refresh starting from the begging (`currentLegGeometryIndex == 0`) and the backend decides to trim the annotations and returns:
```
route: { first leg: [1, 2, 3], second leg: [4, 5]}
```
It did return all annotations for the first leg, but trimmed the second leg.

With the existing logic, we'd replace the annotations in the second leg and cause a mismatch of geometry indices to annotations.

I imagine we need to do something like this:
```suggestion
                } else { 
                  AnnotationsRefresher.getRefreshedAnnotations(
                        routeLeg.annotation(),
                        newAnnotation,
                        0
                    )
                }
```
to pick the all newly available annotations for all legs, and backfill what's necessary.